qiskit/circuit/__init__.py

```diff
@@ -10,38 +10,64 @@
 # copyright notice, and modified files need to carry a notice indicating
 # that they have been altered from the originals.
 
-"""
-========================================
-Quantum Circuits (:mod:`qiskit.circuit`)
-========================================
+r"""
+=============================================
+Quantum circuit model (:mod:`qiskit.circuit`)
+=============================================
 
 .. currentmodule:: qiskit.circuit
 
-Overview
-========
+The fundamental element of quantum computing is the *quantum circuit*.  This is a computational
+routine that can be run, one shot at a time, on a quantum processing unit (QPU).  A circuit will act
```

Comment:
What does "one shot at a time" add to the description? Are you trying to clarify that circuits are not expected to have shot loops?

Reply:
Part of what defines a circuit is, and more crucially, what it _isn't_, is that all the classical data from one circuit can be acquired in one run on a QPU.  I'm specifically trying to exclude the concept of a circuit containing an "expectation value" operation, because

- that's where we split the abstraction level and move to the primitives, and the split of abstraction layers is something we need to keep the software architecture understandable and maintainable
- it would really clash with how all the rest of circuit execution works in the model of being executed on hardware.

The idea being that an implementer of the primitives really wants some software concept of "this is a single shot", and the primitive implementation is what accumulates that into higher-typed classical data, especially since that's going to come with measurement statistics.

It's the accumulation of many shots that I'm trying to completely rule out - the circuit is the abstraction level that defines a single shot.  We _expect_ it to be run many many times in a shot loop during a primitive execution, but if it starts to represent the shot loop itself, it breaks the abstraction layer down between primitives and shots, which will make it harder to architect and work.

Reply:
Ok, that makes sense. Thanks.